sdk/tests/wallet/balance.rs

```diff
@@ -7,11 +7,115 @@ use iota_sdk::{
+        balance_dto.required_storage_deposit.foundry
+    );
+    assert_eq!(
+        balance.required_storage_deposit().nft().to_string(),
+        balance_dto.required_storage_deposit.nft
+    );
+
+    assert_eq!(balance.native_tokens().len(), balance_dto.native_tokens.len());
+    assert_eq!(balance.nfts().len(), balance_dto.nfts.len());
+    assert_eq!(balance.aliases().len(), balance_dto.aliases.len());
+    assert_eq!(balance.foundries().len(), balance_dto.foundries.len());
+    assert_eq!(
+        balance.potentially_locked_outputs().len(),
+        balance_dto.potentially_locked_outputs.len()
+    );
+}
+
+#[test]
+fn balance_add_assign() {
+    let mut balance_1 = AccountBalance::rand_mock();
+    let total_1 = balance_1.base_coin().total();
+    let available_1 = balance_1.base_coin().available();
+    #[cfg(feature = "participation")]
+    let voting_power_1 = balance_1.base_coin().voting_power();
+
+    let sdr_alias_1 = balance_1.required_storage_deposit().alias();
+    let sdr_basic_1 = balance_1.required_storage_deposit().basic();
+    let sdr_foundry_1 = balance_1.required_storage_deposit().foundry();
+    let sdr_nft_1 = balance_1.required_storage_deposit().nft();
+
+    let num_native_tokens_1 = balance_1.native_tokens().len();
+    let num_aliases_1 = balance_1.aliases().len();
+    let num_foundries_1 = balance_1.foundries().len();
+    let num_nfts_1 = balance_1.nfts().len();
+
+    let balance_2 = AccountBalance::rand_mock();
+    let total_2 = balance_2.base_coin().total();
+    let available_2 = balance_2.base_coin().available();
+    #[cfg(feature = "participation")]
+    let voting_power_2 = balance_2.base_coin().voting_power();
+
+    let sdr_alias_2 = balance_2.required_storage_deposit().alias();
+    let sdr_basic_2 = balance_2.required_storage_deposit().basic();
+    let sdr_foundry_2 = balance_2.required_storage_deposit().foundry();
+    let sdr_nft_2 = balance_2.required_storage_deposit().nft();
+
+    let num_native_tokens_2 = balance_2.native_tokens().len();
+    let num_aliases_2 = balance_2.aliases().len();
+    let num_foundries_2 = balance_2.foundries().len();
+    let num_nfts_2 = balance_2.nfts().len();
+
+    balance_1 += balance_2;
+
+    assert_eq!(balance_1.base_coin().total(), total_1 + total_2);
+    assert_eq!(balance_1.base_coin().available(), available_1 + available_2);
+    #[cfg(feature = "participation")]
+    assert_eq!(balance_1.base_coin().voting_power(), voting_power_1 + voting_power_2);
+
+    assert_eq!(balance_1.required_storage_deposit().alias(), sdr_alias_1 + sdr_alias_2);
+    assert_eq!(balance_1.required_storage_deposit().basic(), sdr_basic_1 + sdr_basic_2);
+    assert_eq!(
+        balance_1.required_storage_deposit().foundry(),
+        sdr_foundry_1 + sdr_foundry_2
+    );
+    assert_eq!(balance_1.required_storage_deposit().nft(), sdr_nft_1 + sdr_nft_2);
+
+    assert_eq!(
+        balance_1.native_tokens().len(),
+        num_native_tokens_1 + num_native_tokens_2
+    );
```

Comment:
That looks wrong? Doesn't take into account adding more tokens of the same ID.

Reply:
Each balance has its own random token id, no?

Reply:
I guess I was under the impression that 2 different balances have disjoint set of token ids, like base coin amounts cannot overlap between balances. So that's a wrong assumption, right?

Reply:
Yeah, your different account can have different values of the same native token

Reply:
ah, I see

Reply:
I'll add a test tomorrow that also checks for this kind of scenario.

Reply:
Done